src/mono/mono/profiler/Makefile.am

```diff
@@ -8,6 +8,8 @@ AM_CPPFLAGS = \
 
 glib_libs = $(top_builddir)/mono/eglib/libeglib.la
 
+prof_ldflags = $(CCLDFLAGS)
```

Comment:
I think you need to change `prof_ldflags = ...` to `prof_ldflags += ...` lower down in this file in a couple of places